### PR TITLE
Replaced a dead link for github issue tracker with another github

### PR DESCRIPTION
**Pull request by alan-cugler**
### What does this PR do?
Replaced a dead link for github issue tracker with another github link for the issue tracker and added a link to show more options for closing issues in commit messages.

### What issues does this PR fix or reference?
Didn't bother to make an issue for the deadlink

### Commits signed with GPG?
Yes

### CODE REVIEW

File: doc/topics/development/contributing.rst [hunk 1]
```diff
@@ -100,7 +100,8 @@ Fork a Repo Guide_>`_ and is well worth reading.
     .. note::
 
         If your change fixes a bug or implements a feature already filed in the
-        `issue tracker <GitHub issue tracker>`_, be sure to reference the issue
+        `issue tracker <https://guides.github.com/features/issues/>`_, be sure to
```
Comment (twangboy) [inline doc/topics/development/contributing.rst]: The URL to the Github issue tracker is at the bottom of the document. It should point to salt's issue, not the github guide... as it does at the bottom.

**Review by twangboy — CHANGES_REQUESTED** [review]

File: doc/topics/development/contributing.rst (hunk 1 above)
Comment (alan-cugler) [reply doc/topics/development/contributing.rst]: Ah, I see. Yeah I wasn't understanding what they wanted there. I'll make the link adjustment.

Comment (dwoz) [reply doc/topics/development/contributing.rst]: @alan-cugler ping.

**Comment by Ch3LL** [comment]
@alan-cugler are you okay migrating this PR to the 2018.3 branch since that branch is now the previous docs branch?

**Comment by twangboy** [comment]
@Ch3LL We might just close this unless there's really something wrong with the links

**Comment by dwoz** [comment]
@alan-cugler I am closing due to inactivity, feel free to re-open the PR if this comes up again.

**Comment by alan-cugler** [comment]
I fixed the link and tested it. I request this PR be re-opened for review and added in. I also switched it form 2017.7 to 2018.3.

**Comment by alan-cugler** [comment]
Also, @twangboy I don't think I explained why I had worked on it originally. That URL link is valid but for some reason when clicked on it tries to find a URL page named after the alias and not redirect to the salt github issues. I tinkered with it, and found stripping some of the characters off the alias used for the URL fixed this URL redirect bug.

**Comment by twangboy** [comment]
It's not letting me reopen this. It's saying the `replace-github-dead-link` branch was force-pushed or recreated. You'll probably have to open another PR.

**Comment by twangboy** [comment]
@alan-cugler ^^^

**Comment by alan-cugler** [comment]
@twangboy sounds good, I will do a new PR. it sounds like this was a mistake I made, if anyone knows what I did wrong please let me know so I can avoid the problem in the future!

**Pull request closed without merging**